plugins/locale.js

```diff
@@ -1,30 +1,32 @@
 import { join } from "node:path";
-import { existsSync, readFileSync } from "node:fs";
+import { existsSync } from "node:fs";
 import fp from "fastify-plugin";
-import chalk from "chalk";
-import emojiFlags from "emoji-flags";
+import PathResolver from "../lib/path.js";
 
-// function that takes a locale string and returns the emoji flag for that locale
-// handles formats like en-US, en, en-US-POSIX
-// map countries to languages
-function getFlag(locale) {
-  if (!locale) return "";
-  const countryCode = locale.split("-")[1];
-  if (!countryCode) return "";
-  return `${emojiFlags.countryCode(countryCode).emoji}  `;
-}
+export default fp(async function locale(fastify, { cwd = process.cwd(), locale = "en-US", development }) {
+  const resolver = new PathResolver({ development, cwd });
 
-export default fp(async function locale(fastify, { cwd = process.cwd(), locale = "" }) {
-  const localeFilePath = join(cwd, "locale", locale) + ".json";
-  if (existsSync(localeFilePath)) {
-    try {
-      const translations = JSON.parse(readFileSync(localeFilePath, { encoding: "utf8" }));
-      fastify.decorate("translations", translations);
-      fastify.log.debug(
-        `${getFlag(locale)}${chalk.magenta("translations")}: loaded file "${localeFilePath}" for locale "${locale}"`
-      );
-    } catch (err) {
-      fastify.log.error(`Error reading translation file: ${localeFilePath}`, err);
+  const contentFile = await resolver.resolve("./content");
+  const compiledFileExtension = contentFile.typescript ? ".ts" : ".mjs";
+
+  const localeFilePath = join(cwd, "locales", locale) + compiledFileExtension;
+
+  fastify.decorate("readTranslations", async () => {
+    if (existsSync(localeFilePath)) {
```

Comment:
Use async method fs.stat here instead of the sync version since its being used in a route handler.